sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/ChecksumHandler.cs

```diff
@@ -78,6 +78,14 @@ protected virtual void PreInvoke(IExecutionContext executionContext)
             var request = executionContext.RequestContext.Request;
             var clientConfig = executionContext.RequestContext.ClientConfig;
 
+            // Credentials would be null in the case of anonymous users getting public resources from S3
+            // https://github.com/aws/aws-sdk-net/issues/3696
+            ImmutableCredentials immutableCredentials = executionContext.RequestContext.ImmutableCredentials;
```

Comment:
I'll need to change this in the next V4 sync since we removed `ImmutableCredentials` as part of the SRA.